clients/drcachesim/tools/view.cpp

```diff
@@ -236,6 +247,10 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
              */
             if (TESTANY(OFFLINE_FILE_TYPE_ARCH_REGDEPS, filetype_)) {
                 dr_set_isa_mode(dcontext_.dcontext, DR_ISA_REGDEPS, nullptr);
+                // Ignore the requested syntax: we only support DR style.
+                // XXX i#6942: Should we return an error if the users asks for
+                // another syntax?  Should DR's libraries return an error?
+                disassemble_set_syntax(DR_DISASM_DR);
```

Comment:
No action needed (more a note to myself), but I'm just trying to remember why I didn't put this check and the following `dr_set_isa_mode()` (and now `disassemble_set_syntax()`) in `initialize_stream()` or `parallel_shard_init_stream()`.

Reply:
I think the filetype was not guaranteed to be known at that point is the reason.  Nowadays the scheduler has to read the first few records of the stream in most (all?  should confirm and document better what's guaranteed vs likely) cases and so the filetype is available at init time.

Reply:
Ah yes, that was the reason, thank you Derek.